### DISCUSSION TITLE
Composite icons in IconLayer getIcon function

### DISCUSSION THREAD

**Discussion by jasonsturges** [visgl/deck.gl#7977] [Q&A]
Is it possible to composite multiple images together in an IconLayer's `getIcon()` function?

Attempting to port a Pixi application to Deck.gl, where there are markers that consist of multiple images.

For example this icon consists of a top, bottom, and selection overlay which I'm color transforming - even though the base assets are white, there are multiple images that composite together to complete the indicator:

![icon](https://github.com/visgl/deck.gl/assets/1213591/6526bf2f-9504-4cdb-ba64-e889450426be)

There are several of these at any time overlaid to relay multiple states:

![icons](https://github.com/visgl/deck.gl/assets/1213591/626d3d80-48d4-4275-b69e-8f025ea4364b)

Or what strategy would be recommended here?  Does this have to be three independent icons in the layer?
- Top asset, with color transform
- Bottom asset, with color transform
- Selection highlight

**Comment by c6p** [comment]
I know it is a very old discussion. I have come accros it while searching for another issue.

I found myself in a similar situation. In my case, I had to create borders around images that the client receives from an API endpoint.

The approach I took was to build **SVGs** dynamically and return them **base64**-encoded, which are then rendered by deck.gl.
